<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SimulatorWindow</class>
 <widget class="QMainWindow" name="SimulatorWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1164</width>
    <height>772</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>attoWPU Emulator 0.8.02 alpha</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="attoCoreG">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>5</y>
      <width>556</width>
      <height>186</height>
     </rect>
    </property>
    <property name="title">
     <string>attoCore</string>
    </property>
    <widget class="QGroupBox" name="busG">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>436</width>
       <height>76</height>
      </rect>
     </property>
     <property name="title">
      <string>Bus</string>
     </property>
     <widget class="QLabel" name="LAddr">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>15</y>
        <width>46</width>
        <height>13</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>ADDR</string>
      </property>
     </widget>
     <widget class="QLabel" name="addrB">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>00000000</string>
      </property>
     </widget>
     <widget class="QLabel" name="addrH">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>00000000</string>
      </property>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>55</x>
        <y>15</y>
        <width>16</width>
        <height>56</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
     <widget class="QLabel" name="addrS">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>TO DO</string>
      </property>
     </widget>
     <widget class="QLabel" name="LCtrl">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>15</y>
        <width>46</width>
        <height>13</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>CTRL</string>
      </property>
     </widget>
     <widget class="QLabel" name="ctrlB">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>30</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>00000000</string>
      </property>
     </widget>
     <widget class="QLabel" name="ctrlH">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>40</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>00000000</string>
      </property>
     </widget>
     <widget class="QLabel" name="ctrlS">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>50</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>TO DO</string>
      </property>
     </widget>
     <widget class="Line" name="line_2">
      <property name="geometry">
       <rect>
        <x>115</x>
        <y>15</y>
        <width>16</width>
        <height>56</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
     <widget class="QLabel" name="LData">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>15</y>
        <width>46</width>
        <height>13</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>DATA</string>
      </property>
     </widget>
     <widget class="QLabel" name="dataB">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>30</y>
        <width>196</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>00000000000000000000000000000000</string>
      </property>
     </widget>
     <widget class="QLabel" name="Lhex">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>50</y>
        <width>26</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <italic>false</italic>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>HEX:</string>
      </property>
     </widget>
     <widget class="QLabel" name="dataH">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>50</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>FFFFFFFF</string>
      </property>
     </widget>
     <widget class="QLabel" name="LINT">
      <property name="geometry">
       <rect>
        <x>225</x>
        <y>40</y>
        <width>31</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>UINT:</string>
      </property>
     </widget>
     <widget class="QLabel" name="dataUINT">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>40</y>
        <width>66</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>4294967295</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="LINT_2">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>40</y>
        <width>26</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>INT:</string>
      </property>
     </widget>
     <widget class="QLabel" name="dataINT">
      <property name="geometry">
       <rect>
        <x>155</x>
        <y>40</y>
        <width>66</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>-2147483648</string>
      </property>
     </widget>
     <widget class="QLabel" name="Lfloat">
      <property name="geometry">
       <rect>
        <x>215</x>
        <y>50</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <italic>false</italic>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>F:</string>
      </property>
     </widget>
     <widget class="Line" name="line_3">
      <property name="geometry">
       <rect>
        <x>320</x>
        <y>15</y>
        <width>16</width>
        <height>56</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
     <widget class="QLabel" name="LAjmp">
      <property name="geometry">
       <rect>
        <x>335</x>
        <y>15</y>
        <width>46</width>
        <height>13</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>AJMP</string>
      </property>
     </widget>
     <widget class="QLabel" name="ajmpB">
      <property name="geometry">
       <rect>
        <x>335</x>
        <y>30</y>
        <width>96</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>0000000000000000</string>
      </property>
     </widget>
     <widget class="QLabel" name="dataFLOAT">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>50</y>
        <width>96</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>-7.286752e-044</string>
      </property>
     </widget>
     <widget class="QLabel" name="Lhex_2">
      <property name="geometry">
       <rect>
        <x>335</x>
        <y>40</y>
        <width>26</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <italic>false</italic>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>HEX:</string>
      </property>
     </widget>
     <widget class="QLabel" name="Ldec">
      <property name="geometry">
       <rect>
        <x>335</x>
        <y>50</y>
        <width>26</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <italic>false</italic>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>DEC:</string>
      </property>
     </widget>
     <widget class="QLabel" name="ajmpH">
      <property name="geometry">
       <rect>
        <x>365</x>
        <y>40</y>
        <width>31</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>FFFF</string>
      </property>
     </widget>
     <widget class="QLabel" name="ajmpD">
      <property name="geometry">
       <rect>
        <x>365</x>
        <y>50</y>
        <width>36</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>65535</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="controlG">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>436</width>
       <height>76</height>
      </rect>
     </property>
     <property name="title">
      <string>Control</string>
     </property>
     <widget class="QSlider" name="speedSlider">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>291</width>
        <height>19</height>
       </rect>
      </property>
      <property name="maximum">
       <number>290</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="WPUstop">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>45</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Stop</string>
      </property>
     </widget>
     <widget class="QPushButton" name="WPUstep">
      <property name="geometry">
       <rect>
        <x>55</x>
        <y>45</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Step</string>
      </property>
     </widget>
     <widget class="QPushButton" name="WPU30Hz">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>45</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>30 Hz</string>
      </property>
     </widget>
     <widget class="QPushButton" name="WPU1MHz">
      <property name="geometry">
       <rect>
        <x>145</x>
        <y>45</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>1 MHz</string>
      </property>
     </widget>
     <widget class="QPushButton" name="WPMax">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>45</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Max</string>
      </property>
     </widget>
     <widget class="QLabel" name="WPUspeedL">
      <property name="geometry">
       <rect>
        <x>235</x>
        <y>45</y>
        <width>196</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="text">
       <string>Real: 1000000 Hz</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="speedValue">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>20</y>
        <width>121</width>
        <height>16</height>
       </rect>
      </property>
      <property name="accelerated">
       <bool>false</bool>
      </property>
      <property name="suffix">
       <string> Hz</string>
      </property>
      <property name="decimals">
       <number>2</number>
      </property>
      <property name="maximum">
       <double>10000000.000000000000000</double>
      </property>
      <property name="value">
       <double>0.000000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="unlimitedL">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>325</x>
        <y>20</y>
        <width>86</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Unlimited</string>
      </property>
     </widget>
    </widget>
    <widget class="QLabel" name="processedInstructionsL">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>20</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>TextLabel</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="outputG">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>195</y>
      <width>541</width>
      <height>271</height>
     </rect>
    </property>
    <property name="title">
     <string>Output</string>
    </property>
    <widget class="QGroupBox" name="LEDsG">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>15</y>
       <width>526</width>
       <height>81</height>
      </rect>
     </property>
     <property name="title">
      <string>LED</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <widget class="QLabel" name="row0L">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>15</y>
        <width>41</width>
        <height>14</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>ROW0:</string>
      </property>
     </widget>
     <widget class="QLabel" name="row1L">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>41</width>
        <height>14</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>ROW1:</string>
      </property>
     </widget>
     <widget class="QLabel" name="row2L">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>45</y>
        <width>41</width>
        <height>14</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>ROW2:</string>
      </property>
     </widget>
     <widget class="QLabel" name="row3L">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>41</width>
        <height>14</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>ROW3:</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>95</y>
       <width>146</width>
       <height>171</height>
      </rect>
     </property>
     <property name="title">
      <string>LCD</string>
     </property>
     <widget class="QGraphicsView" name="LCD">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>15</y>
        <width>128</width>
        <height>128</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <kerning>true</kerning>
       </font>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="interactive">
       <bool>false</bool>
      </property>
      <property name="sceneRect">
       <rectf>
        <x>0.000000000000000</x>
        <y>0.000000000000000</y>
        <width>0.000000000000000</width>
        <height>0.000000000000000</height>
       </rectf>
      </property>
     </widget>
     <widget class="QPushButton" name="openHugeLCD">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>145</y>
        <width>126</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Enlarge (new window)</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="geometry">
      <rect>
       <x>155</x>
       <y>95</y>
       <width>351</width>
       <height>91</height>
      </rect>
     </property>
     <property name="title">
      <string>Text Display</string>
     </property>
     <widget class="QPlainTextEdit" name="TextDisplay">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>331</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Lucida Console</family>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="undoRedoEnabled">
       <bool>false</bool>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
      <property name="plainText">
       <string>0123456789012345678901234567890123456789
0123456789012345678901234567890123456789
0123456789012345678901234567890123456789
0123456789012345678901234567890123456789</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="inputG">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>470</y>
      <width>556</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>Input</string>
    </property>
    <widget class="QGroupBox" name="switchesG">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>536</width>
       <height>81</height>
      </rect>
     </property>
     <property name="title">
      <string>Switches</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <widget class="QLabel" name="row0L_2">
      <property name="geometry">
       <rect>
        <x>25</x>
        <y>15</y>
        <width>41</width>
        <height>14</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>ROW0:</string>
      </property>
     </widget>
     <widget class="QLabel" name="row1L_2">
      <property name="geometry">
       <rect>
        <x>25</x>
        <y>30</y>
        <width>41</width>
        <height>14</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>ROW1:</string>
      </property>
     </widget>
     <widget class="QLabel" name="row2L_2">
      <property name="geometry">
       <rect>
        <x>25</x>
        <y>45</y>
        <width>41</width>
        <height>14</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>ROW2:</string>
      </property>
     </widget>
     <widget class="QLabel" name="row3L_2">
      <property name="geometry">
       <rect>
        <x>25</x>
        <y>60</y>
        <width>41</width>
        <height>14</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>ROW3:</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QTabWidget" name="memoryViewer">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>10</y>
      <width>586</width>
      <height>701</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <property name="tabsClosable">
     <bool>false</bool>
    </property>
    <widget class="QWidget" name="attoCodeMemRegion">
     <attribute name="title">
      <string>attoCode</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="RAMRegion">
     <attribute name="title">
      <string>Program+data</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="registerMemoryRegion">
     <attribute name="title">
      <string>Register</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="smallQueueRegion">
     <attribute name="title">
      <string>SmallQueue</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="lcdMemRegion">
     <attribute name="title">
      <string>LCD</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="textDisplayMemRegion">
     <attribute name="title">
      <string>Text Display</string>
     </attribute>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1164</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_HEX"/>
    <addaction name="actionLoad_Source"/>
   </widget>
   <widget class="QMenu" name="menuSimulation">
    <property name="title">
     <string>Emulation</string>
    </property>
    <addaction name="actionSoftReset"/>
    <addaction name="actionHardReset"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout_attoWPU_simulator"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSimulation"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionLoad_HEX">
   <property name="text">
    <string>Load Attocode (HEX)</string>
   </property>
  </action>
  <action name="actionLoad_Source">
   <property name="text">
    <string>Load Source (ATT)</string>
   </property>
  </action>
  <action name="actionSoftReset">
   <property name="text">
    <string>SoftReset</string>
   </property>
  </action>
  <action name="actionHardReset">
   <property name="text">
    <string>HardReset</string>
   </property>
  </action>
  <action name="actionAbout_attoWPU_simulator">
   <property name="text">
    <string>About attoWPU emulator</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
